resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -1428,6 +1535,8 @@ protected int getPriority(Integer override, Map<Class<?>, Integer> contracts, Cl
          Integer p = contracts.get(type);
          if (p != null) return p;
       }
+      // Check for weld proxy.
+      component = component.isSynthetic() ? component.getSuperclass() : component;

```

Comment:
I agree with this approach, so the check applies to any Provider priority retrieval.